flink-connectors/flink-orc/src/main/java/org/apache/flink/orc/OrcUtils.java

```diff
@@ -0,0 +1,2229 @@
+ */
+public class OrcUtils {
+
+	/**
+	 * Convert ORC schema types to Flink types.
+	 *
+	 * @param schema schema of orc file
+	 *
+	 */
+	public static TypeInformation schemaToTypeInfo(TypeDescription schema) {
+		switch (schema.getCategory()) {
+			case BOOLEAN:
+				return BasicTypeInfo.BOOLEAN_TYPE_INFO;
+			case BYTE:
+				return BasicTypeInfo.BYTE_TYPE_INFO;
+			case SHORT:
+				return BasicTypeInfo.SHORT_TYPE_INFO;
+			case INT:
+				return BasicTypeInfo.INT_TYPE_INFO;
+			case LONG:
+				return BasicTypeInfo.LONG_TYPE_INFO;
+			case FLOAT:
+				return BasicTypeInfo.FLOAT_TYPE_INFO;
+			case DOUBLE:
+				return BasicTypeInfo.DOUBLE_TYPE_INFO;
+			case STRING:
+			case CHAR:
+			case VARCHAR:
+				return BasicTypeInfo.STRING_TYPE_INFO;
+			case DATE:
+				return SqlTimeTypeInfo.DATE;
+			case TIMESTAMP:
+				return SqlTimeTypeInfo.TIMESTAMP;
+			case BINARY:
+				return PrimitiveArrayTypeInfo.BYTE_PRIMITIVE_ARRAY_TYPE_INFO;
+			case STRUCT:
+				List<TypeDescription> fieldSchemas = schema.getChildren();
+				TypeInformation[] fieldTypes = new TypeInformation[fieldSchemas.size()];
+				for (int i = 0; i < fieldSchemas.size(); i++) {
+					fieldTypes[i] = schemaToTypeInfo(fieldSchemas.get(i));
+				}
+				String[] fieldNames = schema.getFieldNames().toArray(new String[]{});
+				return new RowTypeInfo(fieldTypes, fieldNames);
+			case LIST:
+				TypeDescription elementSchema = schema.getChildren().get(0);
+				TypeInformation elementType = schemaToTypeInfo(elementSchema);
+				return ObjectArrayTypeInfo.getInfoFor(elementType);
+			case MAP:
+				TypeDescription keySchema = schema.getChildren().get(0);
+				TypeDescription valSchema = schema.getChildren().get(1);
+				TypeInformation keyType = schemaToTypeInfo(keySchema);
+				TypeInformation valType = schemaToTypeInfo(valSchema);
+				return new MapTypeInfo(keyType, valType);
+			case DECIMAL:
+				return BasicTypeInfo.BIG_DEC_TYPE_INFO;
+			case UNION:
+				throw new UnsupportedOperationException("UNION type not supported yet.");
+			default:
+				throw new IllegalArgumentException("Unknown type " + schema);
+		}
+	}
+
+	/**
+	 * Fill rows from orc batch.
+	 *
+	 * @param rows the batch of rows need to be filled
+	 * @param schema schema of orc file
+	 * @param batch current orc batch data used to fill the rows
+	 * @param fieldMapping field mapping
+	 *
+	 */
+	public static void fillRows(Object[] rows, TypeDescription schema, VectorizedRowBatch batch, int[] fieldMapping) {
+
+		int totalRowsInBatch = (int) batch.count();
+
+		List<TypeDescription> fieldTypes = schema.getChildren();
+		for (int outIdx = 0; outIdx < fieldMapping.length; outIdx++) {
+			int inIdx = fieldMapping[outIdx];
+			readField(rows, outIdx, fieldTypes.get(inIdx), batch.cols[inIdx], null, Math.min((int) totalRowsInBatch, rows.length));
+		}
+	}
+
+	private static void readField(Object[] rows, int fieldIdx, TypeDescription schema, ColumnVector vector, long[] lengthVector, int childCount) {
```

Comment:
There are significant of duplications across these helper functions. It requires some refactoring to clean things up.